### ISSUE TITLE
dependents_count seems extremely unreliable

### ISSUE THREAD

**Issue by yschimke** [ossf/criticality_score#55]
https://github.com/tasks/tasks has 340439 for dependents, but seemingly because it's such a generic name.

https://github.com/search?q=github.com%2Ftasks%2Ftasks&type=commits

Most results are basically completed unrelated to the project

**Comment by inferno-chromium** [comment]
There will always be outlier examples for this, we know this is hacky, but currently there is no easy way to get this across all languages, esp c/c++ where no package manager/dependency graphs can be generated. We will add use dependency graphs for non C/C++ languages via package manager in near future. Ideas welcome on how we can fix it.

**Comment by yschimke** [comment]
Arguably I wouldn't call it an outlier, the search query just doesn't mean what you intend it to when the project name is a dictionary word.

**Comment by inferno-chromium** [comment]
> Arguably I wouldn't call it an outlier, the search query just doesn't mean what you intend it to when the project name is a dictionary word.

Yes but is this common. We don't have better ideas atm, but plan to address it somewhat in Issue 31.

**Comment by mikeal** [comment]
Last time I looked into this, the data for this stat was pulled in lazily when the web page for dependents was accessed. That’s probably why it’s so unreliable.

I’ve been asking GitHub for an API for the dependent data for years but it’s not happening. My guess is that this is just really expensive to calculate and keep up to date.

`libraries.io` has much more accurate dep counts https://github.com/mikeal/pkg-dependency-network/blob/master/lib/deps.js

This works for most languages but notably does not work for Go since the service pre-dates gomod.

**Comment by inferno-chromium** [comment]
@mikeal - i am planning to look into using libraries.io data when available. So far, the only issue i see is for projects that don't get released as a package in package managers, there i cannot use libraries.io and it feels weird to fallback to this method. Also, c/c++ projects is an issue.
